src/reducers/math.js

```diff
@@ -58,8 +60,55 @@ const mathReducer = produce((draft, action) => {
             );
           }
         }
+        // update the system collateralization if Art, rate, adapterBalance, spot, or mat changes for any ilk
+        if (
+          [
+            RATE,
+            ILK_ART,
+            ADAPTER_BALANCE,
+            PRICE_WITH_SAFETY_MARGIN,
+            LIQUIDATION_RATIO
+          ].includes(prop)
+        ) {
+          const debts = draft.feeds.map(feed => {
+            if (!draft.raw.ilks[feed.key]) return;
+            const { ilkArt, rate } = draft.raw.ilks[feed.key];
+            if (ilkArt && rate) {
+              return math.debtValue(ilkArt, rate);
+            }
+          });
+          const combinedDebt = debts.reduce((a, b) => {
```

Comment:
style note: if you use `lodash.compact` on `debts` first to take out the blank values, you can skip the `a && b` check here and things get a bit more succinct